### ISSUE TITLE
Dockerised prometheus is stuck at "Starting rule manager..."

### ISSUE THREAD

**Issue by aetherpw** [prometheus/prometheus#14956]
### What did you do?

Run an extended docker image:

```Dockerfile
FROM prom/prometheus:latest

COPY prometheus.yml /etc/prometheus/

```
where prometheus.yml is:

```yml
global:
  scrape_interval: 15s
  evaluation_interval: 15s

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9090"]
```

### What did you expect to see?

A functional dockerised instance of prometheus

### What did you see instead? Under which circumstances?

After "Starting rule manager..." is emitted, prometheus freezes

### System information

Linux 6.1.0-25-amd64 x86_64

### Prometheus version

```text
prometheus, version 2.54.1 (branch: HEAD, revision: e6cfa720fbe6280153fab13090a483dbd40bece3)
  build user:       root@812ffd741951
  build date:       20240827-10:56:41
  go version:       go1.22.6
  platform:         linux/amd64
  tags:             netgo,builtinassets,stringlabels
```

### Prometheus configuration file

```yaml
global:
  scrape_interval: 15s
  evaluation_interval: 15s

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9090"]
```

### Logs

```text
2024-09-21T19:46:40.038096861Z ts=2024-09-21T19:46:40.031Z caller=main.go:601 level=info msg="No time or size retention was set so using the default time retention" duration=15d
2024-09-21T19:46:40.038195040Z ts=2024-09-21T19:46:40.032Z caller=main.go:645 level=info msg="Starting Prometheus Server" mode=server version="(version=2.54.1, branch=HEAD, revision=e6cfa720fbe6280153fab13090a483dbd40bece3)"
2024-09-21T19:46:40.038209466Z ts=2024-09-21T19:46:40.032Z caller=main.go:650 level=info build_context="(go=go1.22.6, platform=linux/amd64, user=root@812ffd741951, date=20240827-10:56:41, tags=netgo,builtinassets,stringlabels)"
2024-09-21T19:46:40.038217839Z ts=2024-09-21T19:46:40.032Z caller=main.go:651 level=info host_details="(Linux 6.1.0-25-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.106-3 (2024-08-26) x86_64 a4877a6fe086 (none))"
2024-09-21T19:46:40.038225438Z ts=2024-09-21T19:46:40.032Z caller=main.go:652 level=info fd_limits="(soft=1048576, hard=1048576)"
2024-09-21T19:46:40.038232687Z ts=2024-09-21T19:46:40.032Z caller=main.go:653 level=info vm_limits="(soft=unlimited, hard=unlimited)"
2024-09-21T19:46:40.049345657Z ts=2024-09-21T19:46:40.047Z caller=web.go:571 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
2024-09-21T19:46:40.050576783Z ts=2024-09-21T19:46:40.050Z caller=main.go:1160 level=info msg="Starting TSDB ..."
2024-09-21T19:46:40.070189713Z ts=2024-09-21T19:46:40.060Z caller=tls_config.go:313 level=info component=web msg="Listening on" address=[::]:9090
2024-09-21T19:46:40.070242824Z ts=2024-09-21T19:46:40.068Z caller=tls_config.go:316 level=info component=web msg="TLS is disabled." http2=false address=[::]:9090
2024-09-21T19:46:40.074407736Z ts=2024-09-21T19:46:40.073Z caller=head.go:626 level=info component=tsdb msg="Replaying on-disk memory mappable chunks if any"
2024-09-21T19:46:40.075378798Z ts=2024-09-21T19:46:40.075Z caller=head.go:713 level=info component=tsdb msg="On-disk memory mappable chunks replay completed" duration=914.853µs
2024-09-21T19:46:40.075439433Z ts=2024-09-21T19:46:40.075Z caller=head.go:721 level=info component=tsdb msg="Replaying WAL, this may take a while"
2024-09-21T19:46:40.095508389Z ts=2024-09-21T19:46:40.095Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=0 maxSegment=7
2024-09-21T19:46:40.106007986Z ts=2024-09-21T19:46:40.104Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=1 maxSegment=7
2024-09-21T19:46:40.143862971Z ts=2024-09-21T19:46:40.143Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=2 maxSegment=7
2024-09-21T19:46:40.163153376Z ts=2024-09-21T19:46:40.162Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=3 maxSegment=7
2024-09-21T19:46:40.172722583Z ts=2024-09-21T19:46:40.171Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=4 maxSegment=7
2024-09-21T19:46:40.181850510Z ts=2024-09-21T19:46:40.180Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=5 maxSegment=7
2024-09-21T19:46:40.185829421Z ts=2024-09-21T19:46:40.185Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=6 maxSegment=7
2024-09-21T19:46:40.186832289Z ts=2024-09-21T19:46:40.186Z caller=head.go:793 level=info component=tsdb msg="WAL segment loaded" segment=7 maxSegment=7
2024-09-21T19:46:40.186903465Z ts=2024-09-21T19:46:40.186Z caller=head.go:830 level=info component=tsdb msg="WAL replay completed" checkpoint_replay_duration=169.981µs wal_replay_duration=110.882959ms wbl_replay_duration=447ns chunk_snapshot_load_duration=0s mmap_chunk_replay_duration=914.853µs total_replay_duration=112.065928ms
2024-09-21T19:46:40.196732387Z ts=2024-09-21T19:46:40.195Z caller=main.go:1181 level=info fs_type=EXT4_SUPER_MAGIC
2024-09-21T19:46:40.196785846Z ts=2024-09-21T19:46:40.195Z caller=main.go:1184 level=info msg="TSDB started"
2024-09-21T19:46:40.196794542Z ts=2024-09-21T19:46:40.195Z caller=main.go:1367 level=info msg="Loading configuration file" filename=/etc/prometheus/prometheus.yml
2024-09-21T19:46:40.197877398Z ts=2024-09-21T19:46:40.197Z caller=main.go:1404 level=info msg="updated GOGC" old=100 new=75
2024-09-21T19:46:40.197945794Z ts=2024-09-21T19:46:40.197Z caller=main.go:1415 level=info msg="Completed loading of configuration file" filename=/etc/prometheus/prometheus.yml totalDuration=1.866723ms db_storage=13.809µs remote_storage=11.718µs web_handler=1.196µs query_engine=20.68µs scrape=751.176µs scrape_sd=154.678µs notify=3.146µs notify_sd=2.767µs rules=12.865µs tracing=34.551µs
2024-09-21T19:46:40.197961011Z ts=2024-09-21T19:46:40.197Z caller=main.go:1145 level=info msg="Server is ready to receive web requests."
2024-09-21T19:46:40.197969040Z ts=2024-09-21T19:46:40.197Z caller=manager.go:164 level=info component="rule manager" msg="Starting rule manager..."
```

**Comment by ful09003** [comment]
@aetherpw :wave: hello! I'd just like to check with you, what does freezing mean in this case?

Once Prometheus finishes its startup tasks (the same set of things you see in those logs), it will at that point be running mostly-silently in the background. You may see more logs output over the course of the process/container lifetime, but usually that is only in the case of errors or Prometheus-specific behaviors like compaction. The logs you've shared are a good, healthy signal.

If the lack of logs is coupled with something like the Prometheus UI not being reachable to you, I'm wondering if you could share the `docker` or `podman` command (assuming it's one of those you're using) you are running? An important thing to note with containers is, unless you specify a local port, you often will not be able to access the containerized application. In my case with Prometheus, that would mean executing something like this: `podman run -p 9090:9090 <image id here>`; the `-p 9090:9090` being vital as it ensures port 9090 on my machine is actually sending traffic to port 9090 of the Prometheus container. Once I do that, I too see the logs cease, but importantly I can load `localhost:9090` in my browser and reach the Prometheus UI.

**Comment by bwplotka** [comment]
Thanks for clarifying. Prometheus is not stuck as @ful09003 mentioned. The word "stuck" means usually to us deadlock/freeze of the process, which is not the case here if you can query the API. It's running, just after initialization there is not a lot of logs.  You can add `--log.level=debug` to enable more (spammy) logs.

Sounds like you expected some data to be scraped, but it's not now? What exactly things you are querying? What metrics you expect? How you target status (/target) UI works? Looks like you only have configured "self" metrics from the Prometheus itself. When you query for `up` or go_goroutines` what do you see?

**Comment by gasguti** [comment]
hello, I had the same problem, I solved updating docker compose from version 2.29.1 to 2.32.1
